mtp_api/apps/notification/management/commands/send_notification_report.py

```diff
@@ -0,0 +1,297 @@
+
+
+def make_local_datetime(date):
+    return timezone.make_aware(datetime.datetime.combine(date, datetime.time.min))
+
+
+def report_period(period_start, period_end):
+    if period_start:
+        period_start = parse_date(period_start)
+        if not period_start:
+            raise CommandError('Cannot parse `--since`')
+        period_start = make_local_datetime(period_start)
+    if period_end:
+        period_end = parse_date(period_end)
+        if not period_end:
+            raise CommandError('Cannot parse `--until`')
+        period_end = make_local_datetime(period_end)
+
+    today = make_local_datetime(timezone.now())
+    if not period_start and not period_end:
+        # default is "last week"
+        period_end = today - datetime.timedelta(days=today.weekday())
+        period_start = period_end - datetime.timedelta(days=7)
+    elif not period_start:
+        # if only end date is specified, take one day
+        period_start = period_end - datetime.timedelta(days=1)
+    elif not period_end:
+        # if only start date is specified, take period up to today
+        period_end = today
+
+    if period_start >= period_end:
+        raise CommandError('`--until` must be after `--since`')
+    elif period_end > period_start + datetime.timedelta(days=7):
+        raise CommandError('Maximum date span is 7 days')
+
+    return period_start, period_end
+
+
+def generate_report(report_path, rule, record_set, serialiser):
+    with report_path.open('w') as f:
+        writer = csv.writer(f)
+        writer.writerow(serialiser.get_headers())
+        count = 0
+        for record in record_set:
+            if rule.applies_to(record) and rule.triggered(record):
+                writer.writerow(serialiser.serialise(record))
+                count += 1
+        return count
+
+
+def create_zip(period_filename, bundle_path):
+    report_paths = bundle_path.glob('*.csv')
+    report_paths = filter(lambda p: p.stat().st_size > 0, report_paths)
+    report_paths = sorted(report_paths)
+    if not report_paths:
+        return None
+
+    zip_path = bundle_path / f'{period_filename}.zip'
+    with zipfile.ZipFile(zip_path, 'w') as z:
+        for path in report_paths:
+            z.write(path, path.name)
+    return zip_path
+
+
+def send_zip(period_description, zip_file, emails):
+    email = AnymailMessage(
+        subject=f'Prisoner money notifications for {period_description}',
+        body=f'''
+OFFICIAL SENSITIVE
+
+Please find attached, the prisoner money notifications reports for {period_description}.
+
+If you have any queries, contact {settings.TEAM_EMAIL}.
+        '''.strip(),
+        from_email=default_from_address(),
+        to=emails,
+        tags=['notifications-report'],
+    )
+    email.attach_file(str(zip_file), mimetype='application/zip')
+    email.send()
+
+
+class Serialiser:
+    def __init__(self, rule):
+        self.rule = rule
+        self.monitored_rule = isinstance(rule, MonitoredRule)
```

Comment:
I would call this `is_monitored_rule` so that it's obvious it's a bool instead of a rule